x-pack/plugins/canvas/server/lib/populate_server_registries.js

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { typesRegistry } from '../../common/lib/types_registry';
+import { functionsRegistry as serverFunctions } from '../../common/lib/functions_registry';
+import { getPluginPaths } from './get_plugin_paths';
+
+const registries = {
+  serverFunctions: serverFunctions,
+  commonFunctions: serverFunctions,
+  types: typesRegistry,
+};
+
+export const populateServerRegistries = (types = []) =>
```

Comment:
Make sure this happens at startup.

Create getServerRegistries function that returns a promise.

Reply:
Make types = Object.keys(registries)